src/en/authors-charm-leadership.md

```diff
@@ -0,0 +1,187 @@
+    the install hook; by propagating it instantly, any running minions can make
+    use of the data and progress immediately, without having to wait for the
+    leader to finish its hook.
+
+    It also means that you can guarantee that a successful `leader-set` call has
+    been reflected in the database, and that all minions will converge towards
+    seeing that value, even if an unexpected error takes down the current hook.
+
+    For both these reasons we strongly recommend that leader settings are always
+    written as a self-consistent group (`leader-set foo=bar baz=qux ping=pong`,
+    rather than `leader-set foo=bar; leader-set baz=qux` etc, in which minions
+    may end up seeing a sandbox in which only `foo` is set to the "correct"
+    value).
+
+  * `leader-get` acts much like relation-get, in that it lets you read string
+    values by key (and expose them in helpful formats), but with the following
+    difference:
+
+      * it reads only from the single leader-settings bucket
+
+    ...and the following key similarity:
+
+      * it presents a sandboxed view of leader-settings data.
+
+    This is necessary, as it is for relation data, because a hook context needs
+    to present *consistent* data; but it means that there's a small extra burden
+    on users of `leader-set`.
+
+## How do I...
+
+### ...share one-shot configuration among units?
+
+Assuming your own implementations of `create_settings` and `valid_settings`, you
+can use the two pseudopython snippets below:
+
+    def set_shared_settings():
+        if is_leader():
+            if !valid_settings(leader_get()):
+                settings = create_settings()
+                leader_set(settings)
+
+    def get_shared_settings():
+        settings = leader_get()
+        if !valid_settings(settings):
+             raise WaitingForLeader()
+        return settings
+
+...which can be used as follows:
+
+  * `set_shared_settings` must be called in `leader-elected`, and may be called
+    anywhere else you like (for example, at the very beginning of `install`, to
+    cause those settings to be ppropagated to minions as soon as possible).
+  * `get_shared_settings` must be called (and handled!) in
+    `leader-settings-changed`, and may also be called at any other time it's
+    convenient; you should always be prepared to catch WaitingForLeader and
+    handle it appropriately (most likely by setting a "waiting" status and
+    exiting without error, to wait for the `leader-settings-changed` which
+    should arrive soon).
+
+### ...share varying configuration among units?
+
+You should be able to use the exact same constructs as above, in the same way;
+you just might want to call `set_shared_settings` in a few more places. If you
+need additional synchronisation, you can use a peer relation to communicate
+minions' acknowledgements back to the leader.
+
+!!! Note: peer relation membership is not guaranteed to match current reality
```

Comment:
I somewhat understand this aside, but it doesn't seem clear if you were coming at this from a fresh charm authors perspective.